src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -984,7 +984,7 @@ private long GetAssemblyId(WriteObjectInfo objectInfo)
             {
                 assemId = 0;
             }
-            else if (assemblyString.Equals(Converter.s_urtAssemblyString))
+            else if (assemblyString.Equals(Converter.s_urtAssemblyString) || assemblyString.Equals(Converter.s_urtAlternativeAssemblyString))
```

Comment:
The stripping logic is already using the correct assembly string. The problem is that some types don't have the TypeForwardedFrom attribute on it like e.g. int[]. int[] is automatically derived from Array which has the TypeForwardedFrom on it (mscorlib) but I have no idea how to set it for int[]. Should we really go this path?...